web/src/main/webapp/xslt/skin/default/skin.xsl

```diff
@@ -2,128 +2,214 @@
+                <li id="map-menu" class="hidden-nojs">
+                  <a title="{$t/map}"
+                     href="{$mapUrl}">
+                    <i class="fa fa-fw fa-globe hidden-sm">&#160;</i>
+                    <span><xsl:value-of select="$t/map"/></span></a>
+                </li>
+              </xsl:if>
+            </ul>
+
+            <xsl:variable name="isAuthenticated"
+                          select="util:isAuthenticated()"/>
+            <xsl:variable name="isSigninEnabled"
+                          select="if (util:getUiConfigurationJsonProperty(/root/request/ui, 'mods.authentication.enabled') = 'false')
+                                    then false()
+                                    else true()"/>
+            <xsl:if test="$isSigninEnabled and not($isAuthenticated) and not($isDisableLoginForm)">
+              <xsl:variable name="signinUrl"
+                            select="if(util:getUiConfigurationJsonProperty(/root/request/ui, 'mods.signin.appUrl'))
+                                    then geonet:updateUrlPlaceholder(util:getUiConfigurationJsonProperty(/root/request/ui, 'mods.signin.appUrl'), /root/gui/nodeId, $lang)
+                                    else concat(/root/gui/nodeUrl, $lang , '/catalog.signin')"/>
+              <ul class="nav navbar-nav navbar-right">
+                <li>
+                  <a href="{$signinUrl}"
+                     title="{$t/signIn}">
+                    <i class="fa fa-sign-in fa-fw">&#160;</i>
+                    <xsl:value-of select="$t/signIn"/>
+                  </a>
+                </li>
+              </ul>
+            </xsl:if>
+          </div>
         </div>
       </div>
-    </div>
 
-    <xsl:if test="/root/search/response">
-      <form action="{$nodeUrl}search"
-            class="form-horizontal" role="form">
-        <div class="row gn-top-search" style="margin:20px">
-          <div class="col-md-offset-3 col-md-1 relative"><b><xsl:value-of select="$t/search"/></b></div>
-          <div class="col-md-5 relative">
-            <div class="gn-form-any input-group input-group-lg">
-              <input type="text"
-                      name="any"
-                      id="gn-any-field"
-                      aria-label="{$t/anyPlaceHolder}"
-                      placeholder="{$t/anyPlaceHolder}"
-                      value="{/root/request/any}"
-                      class="form-control"
-                      autofocus=""/>
-              <div class="input-group-btn">
-                <button type="submit"
-                        class="btn btn-default"
-                        title="{$t/search}">
-                  <i class="fa fa-search">&#160;</i>
-                </button>
-                <a href="{$nodeUrl}search"
-                    class="btn btn-default"
-                    title="{$t/reset}">
-                  <i class="fa fa-times">&#160;</i>
-                </a>
+      <xsl:if test="/root/search/response">
```

Comment:
I'm not sure if this snippet is relevant anymore, the code looks specific to the 3.X versions, for example: https://github.com/geonetwork/core-geonetwork/pull/7015/files#diff-0288f54a93a5cbb90245269c2610c967bea1bfedad167b40f57270c08f9e0cc1R176

In any case it's not new in the PR.